src/test/java/gov/healthit/chpl/aqa/stepDefinitions/ChplAPISteps.java

```diff
@@ -24,4 +34,56 @@ public void verifyContentUnderCertifiedHealthITProductListingAndURL(final String
         assertEquals(actualText, text);
         assertEquals(actualURL, urlLink);
     }
+
+    /**
+     * Assert that title should be CHPL API.
+     * @param title expected as CHPL API
+     */
+    @Then("^CHPL API page title should be \"([^\"]*)\"$")
+    public void chplApiPageTitleShouldBe(final String title) {
+        String chplAPITitle = getDriver().getTitle();
+        Assert.assertEquals(chplAPITitle, title);
+    }
+
+    /**
+     * Click all the controller elements link.
+     * @param arg1 as controller elements
+     */
+    @When("^I click on \"([^\"]*)\" link$")
+    public void iClickOnLink(final String arg1) {
+        String xPath = "//*[text()='" + arg1 + "']";
+        WebElement link = ChplAPIPage.controllerElementName(getDriver(), xPath);
+        ((JavascriptExecutor) getDriver()).executeScript("arguments[0].click();", link);
+    }
+
+    /**
+     * Assert that contents from webpage matches the contents of excel.
+     * @param str as sheet name in the excel
+     */
+    @Then("^list of \"([^\"]*)\" operations should be displayed without deprecated calls$")
+    public void listOfOperationsShouldBeDisplayed(final String str) {
+        String sheetName = null;
+        WebDriver driver = getDriver();
+        // Get element from the webpage into the list
+        List<WebElement> listWebElements = ChplAPIPage.controllerElementList(driver);
+        if (str.equalsIgnoreCase(CORR_ACT_PLN_CTRL)) {
+            sheetName = CORR_ACT_PLN;
+        } else {
+            sheetName = str;
+        }
+        ExcelUtil.setExcelFile(sheetName);
+        // Creating the list to hold all the API listed on the webpage by swagger
+        List<String> apiListFromPage = new ArrayList<>();
+        // Get text of the Web Elements list
+        for (WebElement wElement : listWebElements) {
+            apiListFromPage.add(wElement.getText());
+        }
+        List<String> apiLstExl = new ArrayList<>();
+        // Getting api list from the excel into the list
+        apiLstExl = ExcelUtil.getAllCellData();
+        // Checking if the api from the webpage present in the excel
+        for (String api : apiListFromPage) {
+            assertEquals(apiLstExl.contains(api), true);
```

Comment:
Doesn't this check that the API calls you have in the Excel sheet exist on the page? Where's the check that the ones that used to be deprecated don't exist at all?

Also, instead of `assertEquals(something, true)` it'd make more sense if you used `assertTrue(something)`

Reply:
According to my current logic, we are taking all the APIs from the Webpage related to each controller in a list.  Later, reading all the APIs from the excel sheet considering that excel sheet will have only the non-deprecated APIs. Now, we are iterating each API from the webpage into the API list and then reading from the excel sheet. If there is any deprecated API present in the webpage that's not present in the excel sheet, assert will fail.

In this way, we will make sure that till the time all deprecated APIs are not removed from the webpage the test will keep on failing. **Assuming that excel data contains only the non-deprecated APIs.**

Now,
As per my understanding, the excel or CSV should contain only Deprecated APIs and check if these endpoints are present on the webpage or not.
If they are present, we will consider it as failed because webpage is containing the deprecated APIs and if not, it should pass. 

Please let me know if I'm correct

Reply:
I see what you mean. I hadn't understood the logic that you were using. I think the methodology you have here is now fine, though I still don't like using the Excel file.

Could you do it with scenarios, where one value is the controller, and the second value is a delimited list of API calls? Then they could be stored directly in the Feature file, and easily modified if/when we change API endpoints again

Something like:
```
+--------------------+--------------------------------------------------------------------------------+
|Controller          |API Endpoints                                                                   |
+--------------------+--------------------------------------------------------------------------------+
|Certified Product   |Upload a new certified product; confirm a new certified product; etc...         |
+--------------------+--------------------------------------------------------------------------------+
|Developer           |Create a developer; get a developer; ...                                        |
+--------------------+--------------------------------------------------------------------------------+
|Etc...              |Etc...                                                                          |
+--------------------+--------------------------------------------------------------------------------+
```